packages/mobile/src/components/premium-track-purchase-drawer/PremiumTrackPurchaseDrawer.tsx

```diff
@@ -0,0 +1,182 @@
+
+const messages = {
+  title: 'Complete Purchase',
+  summary: 'Summary',
+  artistCut: 'Artist Cut',
+  audiusCut: 'Audius Cut',
+  always0: 'Always $0',
+  youPay: 'You Pay',
+  price: (price: string) => `$${price}`,
+  payToUnlock: 'Pay-To-Unlock',
+  disclaimer:
+    'By clicking on "Buy", you agree to our Terms of Use. Your purchase will be made in USDC via 3rd party payment provider. Additional payment provider fees may apply. Any remaining USDC balance in your Audius wallet will be applied to this transaction. Once your payment is confirmed, your premium content will be unlocked and available to stream.',
+  buy: (price: string) => `Buy $${price}`
+}
+
+const useStyles = makeStyles(({ spacing, typography, palette }) => ({
+  drawer: {
+    paddingVertical: spacing(6),
+    paddingHorizontal: spacing(4),
+    gap: spacing(6),
+    backgroundColor: palette.white
+  },
+  titleContainer: {
+    ...flexRowCentered(),
+    gap: spacing(2),
+    marginBottom: spacing(2),
+    alignSelf: 'center'
+  },
+  title: {
+    fontSize: typography.fontSize.xl,
+    fontFamily: typography.fontByWeight.heavy,
+    color: palette.neutralLight2,
+    textTransform: 'uppercase',
+    lineHeight: typography.fontSize.xl * 1.25
+  },
+  trackTileContainer: {
+    ...flexRowCentered(),
+    borderColor: palette.neutralLight8,
+    borderWidth: 1,
+    borderRadius: spacing(2),
+    backgroundColor: palette.neutralLight10
+  },
+  trackDetailsContainer: {},
+  summaryContainer: {
+    borderColor: palette.neutralLight8,
+    borderWidth: 1,
+    borderRadius: spacing(1)
+  },
+  summaryRow: {
+    ...flexRowCentered(),
+    justifyContent: 'space-between',
+    paddingVertical: spacing(3),
+    paddingHorizontal: spacing(6),
+    borderBottomColor: palette.neutralLight8,
+    borderBottomWidth: 1
+  },
+  lastRow: {
+    borderBottomWidth: 0
+  },
+  greyRow: {
+    backgroundColor: palette.neutralLight10
+  },
+  summaryTitle: {
+    letterSpacing: 1
+  },
+  payToUnlockTitleContainer: {
+    flexDirection: 'row',
+    alignItems: 'center',
+    justifyContent: 'space-between',
+    gap: spacing(2),
+    marginBottom: spacing(2)
+  },
+  buy: {
+    fontSize: typography.fontSize.large,
+    lineHeight: typography.fontSize.large * 1.5
+  },
+  button: {
+    padding: spacing(4),
+    height: spacing(12)
+  }
+}))
+
+export const PremiumTrackPurchaseDrawer = () => {
+  const styles = useStyles()
+  const { neutralLight2, specialLightGreen1 } = useThemeColors()
+  const { data } = useDrawer('PremiumTrackPurchase')
+  const { trackId } = data
+  const track = useSelector((state) => getTrack(state, { id: trackId }))
+  const user = useSelector((state) => getUser(state, { id: track?.owner_id }))
+
+  const handleConfirmPress = useCallback(() => {
+    console.log('buy button pressed')
+  }, [])
+
+  const { premium_conditions: premiumConditions } = track ?? {}
+
+  if (!track || !isPremiumContentUSDCPurchaseGated(premiumConditions) || !user)
+    return null
```

Comment:
I would normally argue that this component should not even be rendered if the conditions here aren't true (prefer that components receive only valid data). But I would understand if that's a difficult thing to check at the layer where we render this component.